test/e2e/cruise/network/network_test.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2019-2024 The Liqo Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//nolint:gosec // Need to run liqoctl command
+package network
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	liqov1 "github.com/liqotech/liqo/apis/core/v1alpha1"
+	"github.com/liqotech/liqo/pkg/gateway"
+	networkflags "github.com/liqotech/liqo/pkg/liqoctl/test/network/flags"
+	"github.com/liqotech/liqo/test/e2e/testutils/config"
+	"github.com/liqotech/liqo/test/e2e/testutils/tester"
+	"github.com/liqotech/liqo/test/e2e/testutils/util"
+)
+
+const (
+	// clustersRequired is the number of clusters required in this E2E test.
+	clustersRequired = 3
+	// testName is the name of this E2E test.
+	testName = "NETWORK"
+)
+
+func TestE2E(t *testing.T) {
+	util.CheckIfTestIsSkipped(t, clustersRequired, testName)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Liqo E2E Suite")
+}
+
+var (
+	ctx         = context.Background()
+	testContext = tester.GetTester(ctx)
+	interval    = config.Interval
+	timeout     = config.Timeout
+
+	providers     []string
+	consumer      string
+	nodePortNodes = networkflags.NodePortNodesAll
+)
+
+var _ = BeforeSuite(func() {
+	for i := range testContext.Clusters {
+		if testContext.Clusters[i].Role == liqov1.ProviderRole {
+			providers = append(providers, testContext.Clusters[i].KubeconfigPath)
+		}
+	}
+	for i := range testContext.Clusters {
+		if testContext.Clusters[i].Role == liqov1.ConsumerRole {
+			consumer = testContext.Clusters[i].KubeconfigPath
+		}
+	}
```

Comment:
I would break out of the loop after the first consumer is found to be more coeherent with the other tests and for a more predictable behaviour